Building App with SDK 29 Crash on Android 10

**Describe the bug**
Building the App with SDK Tools 29 and BuildTools 29.0.2 will not run on android 10 devices.
The official version of the app i.e "1.10.0" is also built with SDK 29 ,
Hence the app is crashing on all android 10 devices.

**To Reproduce**
1. Use any android 10 device or emulator
2. Install from playStore or run from debug
3. App Crash on SplashScreen

**Expected behavior**
Should Run on Android 10 devices.

**Logs**
```
2019-11-19 05:33:57.502 15158-15158/org.fossasia.phimpme E/AndroidRuntime: FATAL EXCEPTION: main
    Process: org.fossasia.phimpme, PID: 15158
    java.lang.RuntimeException: Unable to start activity ComponentInfo{org.fossasia.phimpme/org.fossasia.phimpme.gallery.activities.SplashScreen}: android.view.InflateException: Binary XML file line #17 in org.fossasia.phimpme:layout/abc_screen_simple: Binary XML file line #17 in org.fossasia.phimpme:layout/abc_screen_simple: Error inflating class androidx.appcompat.widget.FitWindowsLinearLayout
        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3374)
        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3513)
        at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:83)
        at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135)
        at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2109)
        at android.os.Handler.dispatchMessage(Handler.java:107)
        at android.os.Looper.loop(Looper.java:214)
        at android.app.ActivityThread.main(ActivityThread.java:7682)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:516)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:950)
     Caused by: android.view.InflateException: Binary XML file line #17 in org.fossasia.phimpme:layout/abc_screen_simple: Binary XML file line #17 in org.fossasia.phimpme:layout/abc_screen_simple: Error inflating class androidx.appcompat.widget.FitWindowsLinearLayout
     Caused by: android.view.InflateException: Binary XML file line #17 in org.fossasia.phimpme:layout/abc_screen_simple: Error inflating class androidx.appcompat.widget.FitWindowsLinearLayout
     Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.Object java.lang.reflect.Field.get(java.lang.Object)' on a null object reference
        at com.mikepenz.iconics.context.ReflectionUtils.getValue(ReflectionUtils.java:29)
        at com.mikepenz.iconics.context.InternalLayoutInflater.createCustomViewInternal(InternalLayoutInflater.java:155)
        at com.mikepenz.iconics.context.InternalLayoutInflater.access$000(InternalLayoutInflater.java:20)
        at com.mikepenz.iconics.context.InternalLayoutInflater$PrivateWrapperFactory2.onCreateView(InternalLayoutInflater.java:246)
        at android.view.LayoutInflater$FactoryMerger.onCreateView(LayoutInflater.java:239)
        at android.view.LayoutInflater.tryCreateView(LayoutInflater.java:1069)
        at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:997)
        at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:961)
        at android.view.LayoutInflater.inflate(LayoutInflater.java:659)
        at com.mikepenz.iconics.context.InternalLayoutInflater.inflate(InternalLayoutInflater.java:81)
        at android.view.LayoutInflater.inflate(LayoutInflater.java:534)
        at android.view.LayoutInflater.inflate(LayoutInflater.java:481)
        at androidx.appcompat.app.AppCompatDelegateImpl.createSubDecor(AppCompatDelegateImpl.java:773)
        at androidx.appcompat.app.AppCompatDelegateImpl.ensureSubDecor(AppCompatDelegateImpl.java:659)
        at androidx.appcompat.app.AppCompatDelegateImpl.setContentView(AppCompatDelegateImpl.java:552)
        at androidx.appcompat.app.AppCompatActivity.setContentView(AppCompatActivity.java:161)
        at org.fossasia.phimpme.base.BaseActivity.onCreate(BaseActivity.java:53)
        at org.fossasia.phimpme.base.ThemedActivity.onCreate(ThemedActivity.java:40)
        at org.fossasia.phimpme.base.SharedMediaActivity.onCreate(SharedMediaActivity.java:12)
        at org.fossasia.phimpme.gallery.activities.SplashScreen.onCreate(SplashScreen.java:63)
        at android.app.Activity.performCreate(Activity.java:7815)
        at android.app.Activity.performCreate(Activity.java:7804)
        at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1318)
        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3349)
2019-11-19 05:33:57.502 15158-15158/org.fossasia.phimpme E/AndroidRuntime:     at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3513)
        at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:83)
        at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135)
        at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2109)
        at android.os.Handler.dispatchMessage(Handler.java:107)
        at android.os.Looper.loop(Looper.java:214)
        at android.app.ActivityThread.main(ActivityThread.java:7682)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:516)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:950)
```

**GIF Showing Crash of Debug Release using SDK 29**
![ezgif com-video-to-gif](https://user-images.githubusercontent.com/41864109/69104934-ec57ce80-0a8f-11ea-93a1-1efd0d7f1ad6.gif)

**GIF Showing Crash of latest version App from PlayStore**

![ezgif com-video-to-gif (1)](https://user-images.githubusercontent.com/41864109/69104980-0db8ba80-0a90-11ea-8e7d-25968c2fc61b.gif)


**Smartphone Info:**
<!-- Please complete the following information -->
|               | |
|---------------|-|
|Device         |OnePlus7|
|Android Version|Android 10 OxygenOS 10.0.2    |

**Additional context**
The App runs fine on android versions < 10.
It seems to occur because of some outdated dependency libraries.
Since on android sdk 29, androidx has components been made mandatory ,
It might be that one of the library most probably [Android-Iconics](https://github.com/mikepenz/Android-Iconics) is causing the issue ( As checked from the logs ).
Currently the fast work-ground can be to use SDK 28 to build the project.

**Would you like to work on the issue?**
<!-- Please let us know if you can work on it or the issue should be assigned to someone else. -->
- [ ] Yes
- [ ] No
- Other: 
For fast solution I can create a PR to downgrade the sdk to 28 ,
Need your suggestions and reviews.

@codedsun 
`2019-11-22 21:25:11.908 16584-16638/org.fossasia.phimpme E/AndroidRuntime: FATAL EXCEPTION: AsyncTask #1
    Process: org.fossasia.phimpme, PID: 16584
    java.lang.RuntimeException: An error occurred while executing doInBackground()
        at android.os.AsyncTask$4.done(AsyncTask.java:399)
        at java.util.concurrent.FutureTask.finishCompletion(FutureTask.java:383)
        at java.util.concurrent.FutureTask.setException(FutureTask.java:252)
        at java.util.concurrent.FutureTask.run(FutureTask.java:271)
        at android.os.AsyncTask$SerialExecutor$1.run(AsyncTask.java:289)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
        at java.lang.Thread.run(Thread.java:919)
     Caused by: android.database.sqlite.SQLiteException: near "GROUP": syntax error (code 1 SQLITE_ERROR): , while compiling: SELECT parent, bucket_display_name FROM files WHERE ((owner_package_name IN ( 'org.fossasia.phimpme' ) OR media_type=2 OR media_type=4 OR media_type=0 AND mime_type LIKE 'audio/%' OR media_type=3 OR media_type=0 AND mime_type LIKE 'video/%' OR media_type=1 OR media_type=0 AND mime_type LIKE 'image/%') AND (is_pending=0) AND (is_trashed=0)) AND ((media_type=? ) GROUP BY ( parent ))
        at android.database.DatabaseUtils.readExceptionFromParcel(DatabaseUtils.java:184)
        at android.database.DatabaseUtils.readExceptionFromParcel(DatabaseUtils.java:140)
        at android.content.ContentProviderProxy.query(ContentProviderNative.java:437)
        at android.content.ContentResolver.query(ContentResolver.java:962)
        at android.content.ContentResolver.query(ContentResolver.java:890)
        at android.content.ContentResolver.query(ContentResolver.java:846)
        at org.fossasia.phimpme.gallery.data.providers.MediaStoreProvider.getAlbums(MediaStoreProvider.java:100)
        at org.fossasia.phimpme.gallery.data.providers.MediaStoreProvider.getAlbums(MediaStoreProvider.java:26)
        at org.fossasia.phimpme.gallery.data.HandlingAlbums.loadAlbums(HandlingAlbums.java:54)
        at org.fossasia.phimpme.gallery.activities.SplashScreen$PrefetchAlbumsData.doInBackground(SplashScreen.java:223)
        at org.fossasia.phimpme.gallery.activities.SplashScreen$PrefetchAlbumsData.doInBackground(SplashScreen.java:217)
        at android.os.AsyncTask$3.call(AsyncTask.java:378)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at android.os.AsyncTask$SerialExecutor$1.run(AsyncTask.java:289) 
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167) 
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641) 
        at java.lang.Thread.run(Thread.java:919) `
Some non-SDK access warnings are also there ,
`Accessing hidden field Ljava/io/FileDescriptor;->descriptor:I (greylist, JNI, allowed)`
Ref : [https://developer.android.com/about/versions/10/non-sdk-q](https://developer.android.com/about/versions/10/non-sdk-q)

@iamareebjamal - We are not using IconicsContextWrapper, the crash is of InternalLayoutInflater

```
 Caused by: android.view.InflateException: Binary XML file line #17 in org.fossasia.phimpme:layout/abc_screen_simple: Binary XML file line #17 in org.fossasia.phimpme:layout/abc_screen_simple: Error inflating class androidx.appcompat.widget.FitWindowsLinearLayout
     Caused by: android.view.InflateException: Binary XML file line #17 in org.fossasia.phimpme:layout/abc_screen_simple: Error inflating class androidx.appcompat.widget.FitWindowsLinearLayout
     Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.Object java.lang.reflect.Field.get(java.lang.Object)' on a null object reference
        at com.mikepenz.iconics.context.ReflectionUtils.getValue(ReflectionUtils.java:29)
        at com.mikepenz.iconics.context.InternalLayoutInflater.createCustomViewInternal(InternalLayoutInflater.java:155)
        at com.mikepenz.iconics.context.InternalLayoutInflater.access$000(InternalLayoutInflater.java:20)
        at com.mikepenz.iconics.context.InternalLayoutInflater$PrivateWrapperFactory2.onCreateView(InternalLayoutInflater.java:246)
        at android.view.LayoutInflater$FactoryMerger.onCreateView(LayoutInflater.java:239)
        at android.view.LayoutInflater.tryCreateView(LayoutInflater.java:1069)
        at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:997)
        at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:961)
        at android.view.LayoutInflater.inflate(LayoutInflater.java:659)
        at com.mikepenz.iconics.context.InternalLayoutInflater.inflate(InternalLayoutInflater.java:81)
        at android.view.LayoutInflater.inflate(LayoutInflater.java:534)
        at android.view.LayoutInflater.inflate(LayoutInflater.java:481)

```

@iamareebjamal I have upgrarded to the latest version but the app is still crashing on `LFMainActivity`

```2019-12-23 00:14:51.426 13176-13176/org.fossasia.phimpme E/AndroidRuntime: FATAL EXCEPTION: main
    Process: org.fossasia.phimpme, PID: 13176
    android.view.InflateException: Binary XML file line #17: Binary XML file line #17: Error inflating class androidx.appcompat.view.menu.ActionMenuItemView
    Caused by: android.view.InflateException: Binary XML file line #17: Error inflating class androidx.appcompat.view.menu.ActionMenuItemView
    Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'boolean androidx.appcompat.view.menu.MenuItemImpl.showsTextAsAction()' on a null object reference
        at androidx.appcompat.view.menu.ActionMenuItemView.updateTextButtonVisibility(ActionMenuItemView.java:190)
        at androidx.appcompat.view.menu.ActionMenuItemView.setIcon(ActionMenuItemView.java:233)
        at com.mikepenz.iconics.context.IconicsFactory.onViewCreatedInternal(IconicsFactory.kt:55)
        at com.mikepenz.iconics.context.IconicsFactory.onViewCreated(IconicsFactory.kt:39)
        at com.mikepenz.iconics.context.InternalLayoutInflater$PrivateWrapperFactory2.onCreateView(InternalLayoutInflater.kt:233)
        at android.view.LayoutInflater$FactoryMerger.onCreateView(LayoutInflater.java:186)
        at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:811)
        at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:750)
        at android.view.LayoutInflater.inflate(LayoutInflater.java:492)
        at com.mikepenz.iconics.context.InternalLayoutInflater.inflate(InternalLayoutInflater.kt:80)
        at android.view.LayoutInflater.inflate(LayoutInflater.java:423)
        at androidx.appcompat.view.menu.BaseMenuPresenter.createItemView(BaseMenuPresenter.java:167)
        at androidx.appcompat.view.menu.BaseMenuPresenter.getItemView(BaseMenuPresenter.java:186)
        at androidx.appcompat.widget.ActionMenuPresenter.getItemView(ActionMenuPresenter.java:193)
        at androidx.appcompat.widget.ActionMenuPresenter.flagActionItems(ActionMenuPresenter.java:473)
        at androidx.appcompat.view.menu.MenuBuilder.flagActionItems(MenuBuilder.java:1182)
        at androidx.appcompat.view.menu.BaseMenuPresenter.updateMenuView(BaseMenuPresenter.java:96)
        at androidx.appcompat.widget.ActionMenuPresenter.updateMenuView(ActionMenuPresenter.java:226)
        at androidx.appcompat.view.menu.MenuBuilder.dispatchPresenterUpdate(MenuBuilder.java:298)
        at androidx.appcompat.view.menu.MenuBuilder.onItemsChanged(MenuBuilder.java:1069)
        at androidx.appcompat.view.menu.MenuBuilder.startDispatchingItemsChanged(MenuBuilder.java:1096)
        at androidx.appcompat.app.ToolbarActionBar.populateOptionsMenu(ToolbarActionBar.java:461)
        at androidx.appcompat.app.ToolbarActionBar$1.run(ToolbarActionBar.java:56)
        at android.os.Handler.handleCallback(Handler.java:873)
        at android.os.Handler.dispatchMessage(Handler.java:99)
        at android.os.Looper.loop(Looper.java:226)
        at android.app.ActivityThread.main(ActivityThread.java:7179)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:576)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:942)

```

--- a/app/src/main/AndroidManifest.xml
+++ b/app/src/main/AndroidManifest.xml
@@ -42,6 +42,7 @@
         android:name=".MyApplication"
         android:allowBackup="true"
         android:icon="@mipmap/ic_launcher"
+        android:requestLegacyExternalStorage="true"
         android:label="@string/app_name"
         android:largeHeap="true"
         android:supportsRtl="true"

--- a/app/src/main/java/org/fossasia/phimpme/base/ThemedActivity.java
+++ b/app/src/main/java/org/fossasia/phimpme/base/ThemedActivity.java
@@ -18,7 +18,6 @@ import androidx.appcompat.widget.SwitchCompat;
 import androidx.core.content.ContextCompat;
 import androidx.core.graphics.drawable.DrawableCompat;
 import com.mikepenz.iconics.IconicsDrawable;
-import com.mikepenz.iconics.context.IconicsContextWrapper;
 import com.mikepenz.iconics.typeface.IIcon;
 import java.util.ArrayList;
 import org.fossasia.phimpme.R;
@@ -60,12 +59,6 @@ public class ThemedActivity extends BaseActivity {
       setNavigationBarColor(getPrimaryColor());
       setNavBarColor();
     }
-  }
-
-  @Override
-  protected void attachBaseContext(Context newBase) {
-    // NOTE: icons stuff
-    super.attachBaseContext(IconicsContextWrapper.wrap(newBase));
   }
 
   @TargetApi(Build.VERSION_CODES.LOLLIPOP)

--- a/app/src/main/java/org/fossasia/phimpme/gallery/data/providers/MediaStoreProvider.java
+++ b/app/src/main/java/org/fossasia/phimpme/gallery/data/providers/MediaStoreProvider.java
@@ -85,14 +85,10 @@ public class MediaStoreProvider {
           MediaStore.Files.FileColumns.PARENT, MediaStore.Images.Media.BUCKET_DISPLAY_NAME
         };
 
-    String selection, selectionArgs[];
-
-    selection =
-        MediaStore.Files.FileColumns.MEDIA_TYPE
-            + "=? ) GROUP BY ( "
-            + MediaStore.Files.FileColumns.PARENT
-            + " ";
-    selectionArgs = new String[] {String.valueOf(MediaStore.Files.FileColumns.MEDIA_TYPE_IMAGE)};
+    String selection = MediaStore.Files.FileColumns.MEDIA_TYPE + "=? ";
+    String[] selectionArgs =
+        new String[] {String.valueOf(MediaStore.Files.FileColumns.MEDIA_TYPE_IMAGE)};
+    String sortOrder = MediaStore.Files.FileColumns.PARENT;
 
     Cursor cur =
         context
@@ -102,23 +98,26 @@ public class MediaStoreProvider {
                 projection,
                 selection,
                 selectionArgs,
-                null);
+                sortOrder);
 
     if (cur != null) {
       if (cur.moveToFirst()) {
         int idColumn = cur.getColumnIndex(MediaStore.Files.FileColumns.PARENT);
         int nameColumn = cur.getColumnIndex(MediaStore.Images.Media.BUCKET_DISPLAY_NAME);
+        HashSet<Long> idSet = new HashSet<>();
         do {
           Media media = getLastMedia(context, cur.getLong(idColumn));
           if (media != null && media.getPath() != null) {
             String path = StringUtils.getBucketPathByImagePath(media.getPath());
             boolean excluded = isExcluded(path, context);
-            if (!excluded) {
+            long curIdColumn = cur.getLong(idColumn);
+            if (!excluded && !idSet.contains(curIdColumn)) {
+              idSet.add(curIdColumn);
               Album album =
                   new Album(
                       context,
                       path,
-                      cur.getLong(idColumn),
+                      curIdColumn,
                       cur.getString(nameColumn),
                       getAlbumCount(context, cur.getLong(idColumn)));
               if (album.addMedia(getLastMedia(context, album.getId()))) list.add(album);

